compiler/src/dotty/tools/dotc/typer/EtaExpansion.scala

```diff
@@ -136,10 +136,18 @@ abstract class Lifter {
    *     val x0 = pre
    *     x0.f(...)
    *
-   *  unless `pre` is idempotent.
+   *  unless `pre` is idempotent reference, a `this` reference, a literal value, or a or the prefix of an `init` (`New` tree).
+   *
+   *  Note that default arguments will refer to the prefix, we do not want
+   *  to re-evaluate a complex expression each time we access a getter.
    */
   def liftPrefix(defs: mutable.ListBuffer[Tree], tree: Tree)(using Context): Tree =
```

Comment:
Can we get the old behavior when there are no default arguments? We'd need a boolean passed to `liftPrefix` indicating whether the enclosing method has the `HasDefaultParams` flag.

Reply:
I split this definition into `liftNonIdempotentPrefix` and `liftPrefix` instead. Compining them made the documentation to convoluted. `HasDefaultParams` is used to choose which of those methods to call.